external-crates/move/crates/move-package-alt/src/package/lockfile.rs

```diff
@@ -33,7 +33,7 @@ pub struct Lockfile<F: MoveFlavor + fmt::Debug> {
     unpublished: UnpublishedTable<F>,
 
     #[serde(default)]
-    published: BTreeMap<EnvironmentName, Publication<F>>,
+    published: BTreeMap<Located<EnvironmentName>, Publication<F>>,
```

Comment:
I think we should discuss how best to handle locations in lockfiles, because presumably when we create new lock files we will create the lockfile objects before we write them to disk, and it's not clear in this case what we should use for the locations. I think as we've discussed in the other PR we might want a "program generated" location or something, but then we would maybe want to replace this when we serialize out the lockfile?